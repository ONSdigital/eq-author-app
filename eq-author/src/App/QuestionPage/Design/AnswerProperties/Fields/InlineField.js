import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import { Field, Label } from "components/Forms";

const StyledInlineField = styled(Field)`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 0;
`;

const InlineField = ({ id, label, children }) => (
  <StyledInlineField key={id}>
    <Label bold={false} inline htmlFor={id}>
      {label}
    </Label>
    {children}
  </StyledInlineField>
);

InlineField.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired
};

export default InlineField;
