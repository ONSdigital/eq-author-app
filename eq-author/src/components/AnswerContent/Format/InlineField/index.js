import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import { Field, Label } from "components/Forms";

const StyledInlineField = styled(Field)`
  display: flex;
  margin-bottom: 0.2em;
  align-items: center;
`;
const StyledLabel = styled(Label)`
  margin-right: 0.2em;
  align-items: center;
`;

const InlineField = ({ id, label, children, ...otherProps }) => (
  <StyledInlineField key={id} {...otherProps}>
    <StyledLabel bold inline htmlFor={id}>
      {label}
    </StyledLabel>
    {children}
  </StyledInlineField>
);

InlineField.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

export default InlineField;
