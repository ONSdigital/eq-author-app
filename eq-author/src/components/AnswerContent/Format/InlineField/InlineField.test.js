import React from "react";
import { shallow } from "enzyme";

import InlineField from ".";

const createWrapper = (props = {}, render = shallow) => {
  return render(<InlineField {...props} />);
};

describe("InlineLabel", () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      id: "1",
      label: "Label",
      children: <input type="text" />,
    };
    wrapper = createWrapper(props, shallow);
  });

  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
