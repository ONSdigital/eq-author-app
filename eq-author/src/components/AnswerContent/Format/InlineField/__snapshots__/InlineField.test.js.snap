// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineLabel should render 1`] = `
<InlineField__StyledInlineField
  key="1"
>
  <InlineField__StyledLabel
    bold={true}
    htmlFor="1"
    inline={true}
  >
    Label
  </InlineField__StyledLabel>
  <input
    type="text"
  />
</InlineField__StyledInlineField>
`;
